src/vellum/workflows/nodes/core/map_node/tests/test_node.py

```diff
@@ -172,3 +174,81 @@ class FruitMapNode(MapNode):
         ["apple carrot", "apple potato"],
         ["banana carrot", "banana potato"],
     ]
+
+
+def test_map_node_parallel_execution_with_workflow():
+    # TODO: Find a better way to test this such that it represents what a user would see.
+    # https://linear.app/vellum/issue/APO-482/find-a-better-way-to-test-concurrency-with-map-nodes
+    thread_ids = {}
+
+    # GIVEN a series of nodes that simulate work
+    class BaseNode1(BaseNode):
+        item = MapNode.SubworkflowInputs.item
+
+        class Outputs(BaseOutputs):
+            output: str
+            thread_id: int
+
+        def run(self) -> Outputs:
+            current_thread_id = threading.get_ident()
+            thread_ids[self.item] = current_thread_id
+
+            # Simulate work
+            time.sleep(0.5)
+
+            end = time.time()
+            end_str = datetime.datetime.fromtimestamp(end).strftime("%Y-%m-%d %H:%M:%S.%f")
+
+            return self.Outputs(output=end_str, thread_id=current_thread_id)
+
+    class BaseNode2(BaseNode):
+        item = MapNode.SubworkflowInputs.item
+
+        class Outputs(BaseOutputs):
+            output: str
+            thread_id: int
+
+        def run(self) -> Outputs:
+            current_thread_id = threading.get_ident()
+            thread_ids[self.item] = current_thread_id
+
+            # Simulate work
+            time.sleep(0.5)
+
+            end = time.time()
+            end_str = datetime.datetime.fromtimestamp(end).strftime("%Y-%m-%d %H:%M:%S.%f")
+
+            return self.Outputs(output=end_str, thread_id=current_thread_id)
+
+    class FinalNode(BaseNode):
+        item = MapNode.SubworkflowInputs.item
+
+        class Outputs(BaseOutputs):
+            result: float
+            thread_id: int
+
+        def run(self) -> Outputs:
+            current_thread_id = threading.get_ident()
+
+            return self.Outputs(result=float(self.item), thread_id=current_thread_id)
+
+    # AND a workflow that connects these nodes
+    class TestWorkflow(BaseWorkflow[MapNode.SubworkflowInputs, BaseState]):
+        graph = BaseNode1 >> BaseNode2 >> FinalNode
```

Comment:
Is the bug reproducible without BaseNode2 and FinalNode? If so, let's get rid of them